spyrmsd/graph.py

```diff
@@ -1,42 +1,97 @@
-            num_vertices,
-            vertex_property,
-        )
-    except ImportError:
-        raise ImportError("graph_tool or NetworkX libraries not found.")
-
-__all__ = [
-    "graph_from_adjacency_matrix",
-    "match_graphs",
-    "vertex_property",
-    "num_vertices",
-    "num_edges",
-    "lattice",
-    "cycle",
-    "adjacency_matrix_from_atomic_coordinates",
-]
-
-import numpy as np
-
-from spyrmsd import constants
+    pass
+    
+try:
+   from spyrmsd.graphs.nx import (
+        cycle as nx_cycle,
+        graph_from_adjacency_matrix as nx_graph_from_adjacency_matrix,
+        lattice as nx_lattice,
+        match_graphs as nx_match_graphs,
+        num_edges as nx_num_edges,
+        num_vertices as nx_num_vertices,
+        vertex_property as nx_vertex_property,
+    )
+   available_backends.append("networkx")
+except ImportError:
+   pass
+        
+def get_available_backends():
+    return available_backends 
+
+def set_backend(backend):
+
+    ## Get the current backend
+    current_backend = str(os.environ.get("SPYRMSD_BACKEND"))
+  
+    ## Check if the backend is valid
+    if backend.lower() in ["graph_tool", "graphtool", "graph-tool", "graph tool", "gt"]:
+        backend = "graph-tool"
+    elif backend.lower() in ["networkx", "nx"]:
+        backend = "networkx"
+    else:
+        raise ValueError("This backend is not recognized or supported")
+
+    ## Check if we the backend is installed
+    if not backend in available_backends:
+        raise ImportError(f"The {backend} backend doesn't seem to be installed")
+
+    ## Check if we actually need to switch backends
+    if backend == current_backend:
+        print(f"The backend is already {backend}")
+        return
+        
+    global cycle, graph_from_adjacency_matrix, lattice, match_graphs, num_edges, num_vertices, vertex_property
+    
+    if backend == "graph-tool":      
+        
+        cycle = gt_cycle
+        graph_from_adjacency_matrix = gt_graph_from_adjacency_matrix
+        lattice = gt_lattice
+        match_graphs = gt_match_graphs
+        num_edges = gt_num_edges
+        num_vertices = gt_num_vertices
+        vertex_property = gt_vertex_property
+              
+    elif backend == "networkx":
+              
+        cycle = nx_cycle
+        graph_from_adjacency_matrix = nx_graph_from_adjacency_matrix
+        lattice = nx_lattice
+        match_graphs = nx_match_graphs
+        num_edges = nx_num_edges
+        num_vertices = nx_num_vertices
+        vertex_property = nx_vertex_property      
+
+    ## Update the environment variable
+    os.environ["SPYRMSD_BACKEND"] = backend
+
+if len(available_backends) == 0:
+    raise ImportError("No valid backends found. Please ensure atleast Graph-Tool or NetworkX are properly installed")
+else:
+    if os.environ.get("SPYRMSD_BACKEND") is None:
+        ## Set the backend to the first available (preferred) backend         
+        set_backend(backend=available_backends[0])
```

Comment:
```suggestion
if len(_available_backends) == 0:
    raise ImportError("No valid backends found. Please ensure that either graph-tool or NetworkX are installed.")
else:
    if os.environ.get("SPYRMSD_BACKEND") is None:
        ## Set the backend to the first available (preferred) backend         
        set_backend(backend=available_backends[0])
```
See comment above for the environment variable.